nemo/utils/exp_manager.py

```diff
@@ -920,38 +921,53 @@ def nemo_topk_check_previous_run(self):
         self.best_model_score = self.best_k_models[self.best_model_path]
 
     def on_save_checkpoint(self, trainer, pl_module, checkpoint):
-        # output = None
-        output = super().on_save_checkpoint(trainer, pl_module, checkpoint)
-        if not self.always_save_nemo:
+        if hasattr(pl_module, "custom_on_save_checkpoint"):
+            output = pl_module.custom_on_save_checkpoint(
```

Comment:
Don't you need to call the parent on_save_checkpoint() here before calling the custom_on_save_checkpoint? That was the flow before right? Similar to what is below:

`output = super().on_save_checkpoint(trainer, pl_module, checkpoint)`

The parent `custom_on_save_checkpoint()` is really just handling the `.nemo` part differently right?